### ISSUE TITLE
hash storage variant

### ISSUE THREAD

**Issue by s3inlc** [curlyboi/hashtopus#38]
Would it be possible to add the feature to select another type of storage for the normal hashes instead of mysql? Because the server gets really at its limits (as long as it's not a high-end server) when using larger hash lists (for example ~1million hashes).
As I have some experience with using Cassandra, this would be my wish-system to have as hash storage option. If you like the idea I would also be willing to help programming on this as I used Cassandra also with PHP and I also could help with a test environment with Cassandra installed.

**Comment by curlyboi** [comment]
does it follow sql standard?

**Comment by s3inlc** [comment]
Not exactly. It is limited for some requests as Cassandra is more like a key/value store with some additional possibilities to create indexes.

Just to make sure you're understanding me right. I thought of just use Cassandra for the 'hashes' and not for any other tables, so the other tables would just stay MySQL.

**Comment by curlyboi** [comment]
Hmm... I am very often joining regular tables with the hash tables so if I would do this,

I would need to transfer the whole database to Cassandra.

**Comment by s3inlc** [comment]
Too bad, that would not be possible to do all with Cassandra, because for example joins are not possible on Cassandra. I thought perhaps it would be possible to just use it as storage for the hashes because this is the critical part when having large hash lists.

Would it be much work to list me the join statements you are using with the hashtables, or at least give some examples? I will take a look at it, perhaps I will just fork the current repo and play around a little and will look if there could be a workaround for this problem.

**Comment by curlyboi** [comment]
Yeah you can definitely look into the source. The SQL is pretty self-explanatory there.

**Comment by ghost** [comment]
You could organize hashes in a seperate database and organize them by hex value, that way you'd decrease the table size. (For example you order by aa, ab, ac, ad) That'd take 255 tables max.

**Comment by ghost** [comment]
Any updates on this?
